asn1/InterledgerProtocol.asn

```diff
@@ -27,4 +28,21 @@ InterledgerProtocolPaymentMessage ::= SEQUENCE {
     }
 }
 
+InterledgerProtocolError ::= SEQUENCE {
+    -- Standardized error code
+    code IA5String (SIZE (3)),
+    -- Corresponding error name
+    codeName IA5String,
+    -- Participant that originally emitted the error
+    triggeredBy Address,
+    -- Participants that forwarded the error
+    -- TODO: This could be more trouble than it's worth - could get really big etc.
```

Comment:
If connectors did actually include their addresses, you could use a failed payment as a trace route (not sure if that's a good thing or bad)